### PR TITLE
[wasm][debugger]Add hidden attribute support

### PR DESCRIPTION
**Pull request by ilonatommy**
Adds support for System.Diagnostics.DebuggerHiddenAttribute - [#57085](https://github.com/dotnet/runtime/issues/57085).

### CODE REVIEW

File: src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs [hunk 1]
```diff
@@ -1301,6 +1301,11 @@ private async Task SetBreakpoint(SessionId sessionId, DebugStore store, Breakpoi
             {
                 SourceLocation loc = sourceId.First();
                 req.Method = loc.CliLocation.Method;
+                if (req.Method.IsHiddenFromDebugger)
+                {
```
Comment (thaystg) [inline src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs:1305]: You can remove the `{` here

Comment (ilonatommy) [reply src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs]: Resolved with [commit](https://github.com/dotnet/runtime/pull/61143/commits/3deab4b59924da80b828ac016169cc967d2c717e).

**Review by thaystg — APPROVED** [review]

File: src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs [hunk 2]
```diff
@@ -363,6 +364,18 @@ public MethodInfo(AssemblyInfo assembly, MethodDefinitionHandle methodDefHandle,
...
+
+                foreach (var cattr in methodDef.GetCustomAttributes())
+                {
+                    var ctorHandle = asmMetadataReader.GetCustomAttribute(cattr).Constructor;
+                    if (ctorHandle.Kind == HandleKind.MemberReference)
+                    {
+                        var container = asmMetadataReader.GetMemberReference((MemberReferenceHandle)ctorHandle).Parent;
+                        var name = asmMetadataReader.GetString(asmMetadataReader.GetTypeReference((TypeReferenceHandle)container).Name);
+                        if (name == "DebuggerHiddenAttribute")
+                            this.IsHiddenFromDebugger = true;
```
Comment (marek-safar) [inline src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs:376]: is it missing break?

Comment (ilonatommy) [reply src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs:376]: Resolved in [PR](https://github.com/dotnet/runtime/pull/61165).

**Pull request merged**